docs/design/scheduler-core-design.md

```diff
@@ -0,0 +1,216 @@
+<!--
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ -->
+
+# Design of YuniKorn scheduler
+
+## Overall
+
+## Architecture
+
+### Components:
+
+```
+
+                     +---------------+  +--------------+
+                     |K8s Shim       |  |YARN Shim     |
+                     +---------------+  +--------------+
+
+                                +--------------+   +------------+
+                Scheduler-      | GRPC Protocol|   |Go API      |
+                Interface:      +--------------+   +------------+
+   +----------------------------- YuniKorn-Core -------------------------------+
+                        +--------------------+
+                        |    RMProxy         |
+                        +---------+----------+
+                                  |
+                                  |Write Ops                    +----------------+
+    +-------------+               V                            ++Scheduler       |
+    |ConfigWacher +        +---------------+    Allocate       ||   And          |
+    +-------------+        |Scheduler Cache|  <-----------------|                |
+            +---------->   +---------------+    Preempt        ++Preemptor       |
```

Comment:
Its better we keep config out of core area? as anyway it will be a CRD etc later. 
So it may confuse if we put it more tightly coupled.